bootstrap/src/basis/intnb.ml

```diff
@@ -105,15 +105,14 @@ module Make_common (T : I_common) : S_common with type t := usize = struct
       assert (narrow t0 = t0);
       assert (narrow t1 = t1);
       let rel = match T.signed || (Sys.int_size > T.num_bits) with
-        | true -> compare t0 t1
-        | false -> compare (t0 + min_int) (t1 + min_int)
+        | true -> Stdlib.Int.compare t0 t1
+        | false -> Stdlib.Int.compare (t0 + min_int) (t1 + min_int)
```

Comment:
Oh nice, that compare function you linked to is a great way to implement comparison, but I didn't even try it because I expected the compiler to poorly optimize it. I'll give it a try!

Reply:
I went down the road of using `Obj.magic`, but there was still the problem of converting `int` to `Cmp.t`.  More `Obj.magic` to solved that.  In the end though, I just used what I learned along the way to implement efficient native functions.

Reply:
Did the changes make a difference in your benchmarks?

Reply:
The changes relative to the version you reviewed did improve performance a bit.  But I saw no measurable difference between the final `Obj.magic`-based implementation and the native function version.